### DISCUSSION TITLE
No Zones available

### DISCUSSION THREAD

**Discussion by witc-git** [rainepretorius/olarm-ha-integration#81] [Q&A] [answered]
I have just replaced my old Inim alarm with an IDS X64 and an Olarm to use with HA, I got the integration working, and I can use some of the arm/disarm buttons, though others (such as the quick access ones in the alarm panel do not do anything, and in my entities list I see a lot of PGMs that I cannot do anything with (only 4 is connected and its pulse, so it makes sense), I also have a lot of Ukeys that I'm not sure what they are yet, but I have no zones whatsoever. Is there something in the physical setup that needs to happen in order for the zones to pickup on HA (they do pickup & work in the Olarm app). Do I have to manually create the zones? If so, how do I link them with the PIR/magnets on the zones?

**Comment by rainepretorius** [comment]
Can you send the integration diagnostic file?

**Reply by witc-git** [reply]
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.9.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.5",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Africa/Johannesburg",
    "os_name": "Linux",
    "os_version": "6.1.45",
    "supervisor": "2023.10.0",
    "host_os": "Home Assistant OS 10.5",
    "docker_version": "23.0.6",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "olarm_sensors": {
      "version": "2.2.4",
      "requirements": [
        "aiohttp"
      ]
    },
    "hacs": {
      "version": "1.33.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "solcast_solar": {
      "version": "4.0.8",
      "requirements": [
        "aiohttp>=3.6.2",
        "datetime>=4.3",
        "isodate>=0.6.1"
      ]
    },
    "luxpower": {
      "version": "4.2.6",
      "requirements": []
    }
  },
  "integration_manifest": {
    "domain": "olarm_sensors",
    "name": "Olarm Sensors",
    "codeowners": [
      "@rainepretorius"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/rainepretorius/olarm-ha-integration/blob/main/README.md",
    "iot_class": "cloud_polling",
    "issue_tracker": "https://github.com/rainepretorius/olarm-ha-integration/issues",
    "quality_scale": "silver",
    "requirements": [
      "aiohttp"
    ],
    "version": "2.2.4",
    "is_built_in": false
  },
  "data": {
    "version": "2.2.4",
    "config": {
      "Home": {
        "auth_success": true,
        "error": null,
        "code_required": true
      },
      "scan_interval": 10
    },
    "errors": [],
    "enabled_devices": [
      "Home"
    ],
    "amount_of_enabled_devices": 1,
    "all_devices": [
      {
        "deviceId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
        "deviceName": "Home",
        "deviceAlarmType": "ids_x64",
        "deviceAlarmTypeDetail": null,
        "deviceStatus": "online",
        "deviceState": {
          "areas": [
            "disarm",
            "disarm",
            "disarm",
            "disarm",
            "disarm",
            "disarm",
            "disarm",
            "disarm"
          ],
          "pgm": [
            "c",
            "c",
            "c",
            "c",
            "c",
            "c",
            "c",
            "c"
          ],
          "pgmOb": [
            "c",
            "c"
          ],
          "ukeys": [],
          "power": {
            "AC": "1",
            "Batt": "1"
          }
        },
        "deviceProfile": {
          "ver": 3,
          "areasLimit": 2,
          "areasLabels": [
            "Alarm",
            "Gate"
          ],
          "zonesLimit": 16,
          "pgmLimit": 8,
          "pgmControl": [
            "000",
            "000",
            "000",
            "000",
            "000",
            "000",
            "000",
            "000"
          ],
          "ukeysLimit": 16,
          "fenceZoneLimit": 0,
          "fenceGateLimit": 0
        },
        "deviceTimezone": "Africa/Harare",
        "deviceFirmware": "3004.00500900000"
      }
    ],
    "amount_of_total_devices": 1,
    "uuid": "1696520223157e1854439814f558877957d751d75ed2.2.4",
    "entry_id": "57e1854439814f558877957d751d75ed"
  }
}
Is that the right diagnostic?
Area 1 named Alarm is the actual alarm where the zones live, Area 2 named Gate is a pulse added to PGM4 to trigger a gate to open. I can Arm Home, Arm Away and Disarm the whole alarm from the info window, but not via shortcuts, I also cannot bypass individual zones, however I did just discover the Bypass Home option (switch.bypass_home) which successfully bypasses the door magnet on Zone 1, but nothing else. (in Sensors section I only see the Home status, Powered by AC status and Powered by Battery status)

**Reply by rainepretorius** [reply, marked as answer]
> {
>   "home_assistant": {
>     "installation_type": "Home Assistant OS",
>     "version": "2023.9.3",
>     "dev": false,
>     "hassio": true,
>     "virtualenv": false,
>     "python_version": "3.11.5",
>     "docker": true,
>     "arch": "x86_64",
>     "timezone": "Africa/Johannesburg",
>     "os_name": "Linux",
>     "os_version": "6.1.45",
>     "supervisor": "2023.10.0",
>     "host_os": "Home Assistant OS 10.5",
>     "docker_version": "23.0.6",
>     "chassis": "vm",
>     "run_as_root": true
>   },
>   "custom_components": {
>     "olarm_sensors": {
>       "version": "2.2.4",
>       "requirements": [
>         "aiohttp"
>       ]
>     },
>     "hacs": {
>       "version": "1.33.0",
>       "requirements": [
>         "aiogithubapi>=22.10.1"
>       ]
>     },
>     "solcast_solar": {
>       "version": "4.0.8",
>       "requirements": [
>         "aiohttp>=3.6.2",
>         "datetime>=4.3",
>         "isodate>=0.6.1"
>       ]
>     },
>     "luxpower": {
>       "version": "4.2.6",
>       "requirements": []
>     }
>   },
>   "integration_manifest": {
>     "domain": "olarm_sensors",
>     "name": "Olarm Sensors",
>     "codeowners": [
>       "@rainepretorius"
>     ],
>     "config_flow": true,
>     "dependencies": [],
>     "documentation": "https://github.com/rainepretorius/olarm-ha-integration/blob/main/README.md",
>     "iot_class": "cloud_polling",
>     "issue_tracker": "https://github.com/rainepretorius/olarm-ha-integration/issues",
>     "quality_scale": "silver",
>     "requirements": [
>       "aiohttp"
>     ],
>     "version": "2.2.4",
>     "is_built_in": false
>   },
>   "data": {
>     "version": "2.2.4",
>     "config": {
>       "Home": {
>         "auth_success": true,
>         "error": null,
>         "code_required": true
>       },
>       "scan_interval": 10
>     },
>     "errors": [],
>     "enabled_devices": [
>       "Home"
>     ],
>     "amount_of_enabled_devices": 1,
>     "all_devices": [
>       {
>         "deviceId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
>         "deviceName": "Home",
>         "deviceAlarmType": "ids_x64",
>         "deviceAlarmTypeDetail": null,
>         "deviceStatus": "online",
>         "deviceState": {
>           "areas": [
>             "disarm",
>             "disarm",
>             "disarm",
>             "disarm",
>             "disarm",
>             "disarm",
>             "disarm",
>             "disarm"
>           ],
>           "pgm": [
>             "c",
>             "c",
>             "c",
>             "c",
>             "c",
>             "c",
>             "c",
>             "c"
>           ],
>           "pgmOb": [
>             "c",
>             "c"
>           ],
>           "ukeys": [],
>           "power": {
>             "AC": "1",
>             "Batt": "1"
>           }
>         },
>         "deviceProfile": {
>           "ver": 3,
>           "areasLimit": 2,
>           "areasLabels": [
>             "Alarm",
>             "Gate"
>           ],
>           "zonesLimit": 16,
>           "pgmLimit": 8,
>           "pgmControl": [
>             "000",
>             "000",
>             "000",
>             "000",
>             "000",
>             "000",
>             "000",
>             "000"
>           ],
>           "ukeysLimit": 16,
>           "fenceZoneLimit": 0,
>           "fenceGateLimit": 0
>         },
>         "deviceTimezone": "Africa/Harare",
>         "deviceFirmware": "3004.00500900000"
>       }
>     ],
>     "amount_of_total_devices": 1,
>     "uuid": "1696520223157e1854439814f558877957d751d75ed2.2.4",
>     "entry_id": "57e1854439814f558877957d751d75ed"
>   }
> }
> Is that the right diagnostic?
> Area 1 named Alarm is the actual alarm where the zones live, Area 2 named Gate is a pulse added to PGM4 to trigger a gate to open. I can Arm Home, Arm Away and Disarm the whole alarm from the info window, but not via shortcuts, I also cannot bypass individual zones, however I did just discover the Bypass Home option (switch.bypass_home) which successfully bypasses the door magnet on Zone 1, but nothing else. (in Sensors section I only see the Home status, Powered by AC status and Powered by Battery status)

Yes it is. It seems that Olarm is not sending your zones. Have you done a restart?

**Comment by rainepretorius** [comment]
> I have just replaced my old Inim alarm with an IDS X64 and an Olarm to use with HA, I got the integration working, and I can use some of the arm/disarm buttons, though others (such as the quick access ones in the alarm panel do not do anything, and in my entities list I see a lot of PGMs that I cannot do anything with (only 4 is connected and its pulse, so it makes sense), I also have a lot of Ukeys that I'm not sure what they are yet, but I have no zones whatsoever. Is there something in the physical setup that needs to happen in order for the zones to pickup on HA (they do pickup & work in the Olarm app). Do I have to manually create the zones? If so, how do I link them with the PIR/magnets on the zones?

For the PGM's you can set the amount under Device Setup in the Olarm App

**Reply by witc-git** [reply]
I rebooted HA after installing the integration but this was before the Olarm was online, after a second reboot I see zones and the expected sensors. Thank you for the guidance.

**Reply by rainepretorius** [reply]
> I rebooted HA after installing the integration but this was before the Olarm was online, after a second reboot I see zones and the expected sensors. Thank you for the guidance.

A, I see. As the Ilarm was offline, the integration was not able to get the zone data.
